### DISCUSSION TITLE
Open source hardware Zigbee devices/projects by  DIYRuZ (modkam.ru) members

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#646] [General]
FYI, I don't read Russian but have stumbled on many Zigbee projects from the DIYRuZ / modkam.ru members the last few years and as they post mostly post in the Zigbee2MQTT community I thought might be of interest to hardware developers targeting Zigpy too.

https://github.com/diyruz

https://modkam.ru

I for one would not mind at all seeing versions of those Zigbee devices/projects available as complete kits and/or finished products available for purchase. 

Some examples of [DIYRuZ](https://github.com/diyruz
) ([modkam.ru](https://modkam.ru)) Zigbee based open-source hardware devices and projects: 

DIYRuZ Soil Moisture Sensor Flower  = ZigbeeID: ["DIYRuZ_Flower"]
https://github.com/diyruz/flower
https://modkam.ru/?p=1700
https://zigbee.blakadder.com/DIYRuZ_Flower.html

![image](https://user-images.githubusercontent.com/6320001/105706626-0ef63a80-5f12-11eb-9adb-9f1b41ab5d22.png)

DIYRuZ Air Quality Sensor AirSense  = ZigbeeID: ["DIYRuZ_AirSense"]
https://github.com/diyruz/AirSense
https://modkam.ru/?p=1715
https://zigbee.blakadder.com/DIYRuZ_Airsense.html

![image](https://user-images.githubusercontent.com/6320001/105706922-78764900-5f12-11eb-86cb-5f7d93c37e3f.png)

DIYRuZ DiY FreePad - Customisable 8/12/20 Button Keypad   = ZigbeeID: ["DIYRuZ_FreePad", "FreePadLeTV8"]
https://github.com/diyruz/freepad
https://modkam.ru/?p=1264
https://zigbee.blakadder.com/DIYRuZ_FreePad.html

![image](https://user-images.githubusercontent.com/6320001/105707085-afe4f580-5f12-11eb-8942-be75fe2221e2.png) 
![image](https://user-images.githubusercontent.com/6320001/105707116-b8d5c700-5f12-11eb-909b-d5f0034988c4.png)

DIYRuZ DIY Geiger Counter Geiger  = Zigbee ID: ["DIYRuZ_Geiger"]
https://github.com/diyruz/geiger
https://modkam.ru/?p=1591
https://zigbee.blakadder.com/DIYRuZ_Geiger.html

![image](https://user-images.githubusercontent.com/6320001/105707233-e9b5fc00-5f12-11eb-8041-8470c8956056.png)

DIYRuZ DIY Reed - Zigbee Doorbell Button
https://github.com/diyruz/reed

![image](https://user-images.githubusercontent.com/6320001/105707738-ab6d0c80-5f13-11eb-9abd-77bb3a296e0e.png)

Zigbee 3.0 Router and Single Relay Switch + a 1-wire sensor firmware for CC2530 like Sonoff BASIC ZBR3 = ZigbeeID: ["DIYRuZ_RT"]
https://github.com/diyruz/diyruz_rt
https://habr.com/ru/company/iobroker/blog/495926/
https://zigbee.blakadder.com/DIYRuZ_RT.html

![image](https://user-images.githubusercontent.com/6320001/105706726-33521700-5f12-11eb-9a74-1a741af58fb3.png)

DIYRuZ DIY SensBoard - CC2652 based universal sensor board
https://github.com/diyruz/SensBoard

DIYRuZ DIY 4 Channel Relay + 1 Output R4_5  = ZigbeeID: ["DIYRuZ_R4_5"]
https://zigbee.blakadder.com/DIYRuZ_R4_5.html

![image](https://user-images.githubusercontent.com/6320001/105708526-e3288400-5f14-11eb-8661-c73406ef7f45.png)

DIYRuZ DIY 8 Channel Relay R8_8  = ZigbeeID: ["DIYRuZ_R8_8"]
https://zigbee.blakadder.com/DIYRuZ_R8_8.html

![image](https://user-images.githubusercontent.com/6320001/105708549-eb80bf00-5f14-11eb-822e-374f994d940e.png)

DIYRuZ Collection of reusable source code components for Texas Instruments Z-Stack 3 (Z-Stack 3.0.2) Zigbee firmware:
https://github.com/diyruz/zstack-lib

PS: DIYRuZ also share members with the people working on the Open Lumi (OpenLumi) project that is discussed seperatly in https://github.com/zigpy/zigpy/discussions/644

**Comment by MattWestb** [comment]
Some ideas its good but i dont like putting my time and money on one old CC-253X hard and software then its one dead platform.

Sone user with experience of cooking professional devices was trying the NXP SDK but it was too much hacking to getting all pieces together good and need very much time to getting on working product.
Good with it is you can using Xiaomi devices and flashing with your customer firmware for your projects.

The same user was getting on the Silabs train and saying its much easier getting one project up and running and also implanting custom code in the project.

The good with EZSP is you have IKEA. New Sonoff ZB and tuya devices out there that is pretty cheap for base for you cooked firmware and dont need doing the complete soft and hardware for getting it working.

One very nice device combination with on PI (Zero) ESP to sending and receiving commands in one zigbee network is one _**Zigbee Control bridge**_ from [ZigBee Lighting & Occupancy Device Specification](https://zigbeealliance.org/wp-content/uploads/2019/11/docs-15-0014-05-0plo-Lighting-OccupancyDevice-Specification-V1.0.pdf) chapter 25 Device ID: 0x0840.
Its one "universal Zigbee Light Link Remote" that sending and receiving all commands to ZLL devices in your zigbee network if you have the CLI to one comport. I think its can working with other profiles if you is implanting more endpoints for those cluster you like.

I can thinking having one IKEA module with one WeMos D1 Mini USB powered that is loaded with Tasmota or ESPHome and can sending autonomous commands like "if the zigbee light sensor is having more the 100 Lux put the "spinate light" of if the if the sun is above the horizon (was having 2 hours of sun in december so need extra help for the Magic blue basilikum to survive the winter) and the RGB stripes in the living room windows on then the 3° before the sun is going down until 22.00" and so on.

Cheap easy doing hardware and its 110% autonome if WiFi network is down and SD-card is broken as the ZLL devices is constructed.

So one cheap EZSP dev kit is on my wishing list for trying cooking custom firmware for good hardware.

**Reply by Hedda** [reply]
> Some ideas its good but i dont like putting my time and money on one old CC253X hard and software then its one dead platform.

FYI, all those projects also work with Z-Stack 3 on CC2652 and CC1352 as well, so they are not only for CC2530, CC2531, and CC2538.
